libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/KotlinDependenciesManagement.kt

```diff
@@ -192,7 +192,7 @@ private fun chooseAndAddStdlibDependency(
 
         // If the exact same module is added in the source sets hierarchy, possibly even with a different scope, we don't add it
         val moduleAddedInHierarchy = hierarchySourceSetsDependencyConfigurations.any {
-            it.allDependencies.any { dependency -> dependency.group == KOTLIN_MODULE_GROUP && dependency.name == stdlibModuleName }
+            it.allExternalDependencies().any { dependency -> dependency.group == KOTLIN_MODULE_GROUP && dependency.name == stdlibModuleName }
```

Comment:
According to Gradle folks:
> you can iterate over `allDependencies` and query any of the dependencies. However, `ProjectDependency.getGroup()`  and `ProjectDependency.getVersion()` will fail as these access the state of the target project.

Current approach excludes `FileCollectionDependency` and while adding stdlib jar is low probability, it still may happen. How about filter out only `ProjectDependency`s?
```kt
it.allDependencies.matching { it !is ProjectDependency }
```

Reply:
Done